spec/unit/matrixrtc/RoomAndToDeviceTransport.spec.ts

```diff
@@ -0,0 +1,156 @@
+import { EventType, type IRoomTimelineData, type Room, RoomEvent, type MatrixClient } from "../../../src";
+import type { ToDeviceKeyTransport } from "../../../src/matrixrtc/ToDeviceKeyTransport.ts";
+import {
+    getMockClientWithEventEmitter,
+    mockClientMethodsEvents,
+    mockClientMethodsUser,
+} from "../../test-utils/client.ts";
+import { type Statistics } from "../../../src/matrixrtc";
+import { KeyTransportEvents } from "../../../src/matrixrtc/IKeyTransport.ts";
+import { type Logger } from "../../../src/logger.ts";
+import { RoomAndToDeviceEvents, RoomAndToDeviceTransport } from "../../../src/matrixrtc/RoomAndToDeviceKeyTransport.ts";
+import type { RoomKeyTransport } from "../../../src/matrixrtc/RoomKeyTransport.ts";
+
+describe("RoomAndToDeviceTransport", () => {
+    const roomId = "!room:id";
+
+    let mockClient: Mocked<MatrixClient>;
+    let statistics: Statistics;
+    let mockLogger: Mocked<Logger>;
+    let transport: RoomAndToDeviceTransport;
+    let mockRoom: Room;
+    let sendEventMock: jest.Mock;
+    function getToDeviceTransport(transport: RoomAndToDeviceTransport): ToDeviceKeyTransport {
+        return (transport as unknown as any).toDeviceTransport as ToDeviceKeyTransport;
+    }
+    function getRoomTransport(transport: RoomAndToDeviceTransport) {
+        return (transport as unknown as any).roomKeyTransport as RoomKeyTransport;
+    }
+    beforeEach(() => {
+        sendEventMock = jest.fn();
+        mockClient = getMockClientWithEventEmitter({
+            encryptAndSendToDevice: jest.fn(),
+            getDeviceId: jest.fn().mockReturnValue("MYDEVICE"),
+            ...mockClientMethodsEvents(),
+            ...mockClientMethodsUser("@alice:example.org"),
+            sendEvent: sendEventMock,
+        });
+        mockRoom = makeMockRoom([]);
+        mockLogger = {
+            debug: jest.fn(),
+            warn: jest.fn(),
+            getChild: jest.fn(),
+        } as unknown as Mocked<Logger>;
+        mockLogger.getChild.mockReturnValue(mockLogger);
+        statistics = {
+            counters: {
+                roomEventEncryptionKeysSent: 0,
+                roomEventEncryptionKeysReceived: 0,
+            },
+            totals: {
+                roomEventEncryptionKeysReceivedTotalAge: 0,
+            },
+        };
+        transport = new RoomAndToDeviceTransport(
+            "@alice:example.org",
+            "MYDEVICE",
+            mockRoom,
+            mockClient,
+            statistics,
+            mockLogger,
+        );
+    });
+
+    it("should enable to device transport when starting", () => {
+        transport.start();
+        expect(transport.enabled.room).toBeFalsy();
+        expect(transport.enabled.toDevice).toBeTruthy();
+    });
+    it("only sends to device keys when sending a key", async () => {
+        transport.start();
+        const toDeviceSpy = jest.spyOn(getToDeviceTransport(transport), "sendKey");
```

Comment:
Instead of doing that, wouldn't it be better to jsut ensure which of `encryptAndSendToDevice` and `sendEvent` mock is called?

Reply:
I was wondering the same. It depends how much we want to test here.
I am a fan of smaller "unit" where there are clear interfaces for the classes public api.
When using encryptAndSendToDevice we basically also test the toDeviceTransport which could be tested on its own.
This way the tests are less fragile to changes elsewhere in the codebase (not in the `RoomAndToDeviceTransport`)